src/TestingDevice.java

```diff
@@ -151,7 +151,10 @@ public TestingDevice(int size, int bitsPerPeek) {
         boolean allTrue = true;
         boolean allFalse = true;
         for (int i = 0; i < size; i++) {
-            if (Math.random() > 0.5) { initialBits[i] = true; allFalse = false; }
+            if (Math.random() > 0.5) {
```

Comment:
I can't even. I am not sure how this got slipped into my code.

I try to avoid this kind of codestyle as a rule of thumb. Fixed!